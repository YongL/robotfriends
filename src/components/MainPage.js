import React, { Component } from 'react';

import CardList from './cardList'
import Searchbox from './searchbox';
import Scroll from './scroll';
import './MainPage.css';
import ErrorBoundary from './errorBoundary';
import Header from './header';


class MainPage extends Component {
  componentDidMount() {
    this.props.onRequestRobots();
  }

  filterRobots = (robots, searchField) =>
    robots.filter(robot => {
      return robot.name.toLowerCase().includes(searchField.toLowerCase());
    });

  render() {
    const { robots, searchField, onSearchChange, isPending } = this.props;
    // const filteredRobots = robots.filter(robot => {
    //   return robot.name.toLowerCase().includes(searchField.toLowerCase());
    // })
    return (
      <div className='tc'>
        <Header />
        <Searchbox searchChange={onSearchChange}/>
        <Scroll>
          { isPending ? <h1>Loading</h1> :
            <ErrorBoundary>
              <CardList robots={this.filterRobots(robots, searchField)} />
            </ErrorBoundary>
          }
        </Scroll>
      </div>
    );
  }
}

export default MainPage;
