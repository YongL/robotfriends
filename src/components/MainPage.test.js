import { shallow } from 'enzyme';
import React from 'react'; // import this because it is jsx
import MainPage from './MainPage';

describe('test main page', () => {

    const mockProps = {
        onRequestRobots: jest.fn(),
        robots: [],
        searchField: '',
        isPending: false
    };
    let wrapper;
    beforeEach(() => {
        wrapper = shallow(<MainPage {...mockProps}/>);
    });

    it('render main page', () => {
        expect(wrapper).toMatchSnapshot();
    })

    it('filter robots correctly', () => {
        expect(wrapper.instance().filterRobots([])).toEqual([]);
    });
})

// jest do snaphot testing, which snapshot the codes (string copy not image)

