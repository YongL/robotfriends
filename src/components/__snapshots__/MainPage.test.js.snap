// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test main page render main page 1`] = `
<div
  className="tc"
>
  <Header />
  <Searchbox />
  <Scroll>
    <ErrorBoundary>
      <CardList
        robots={Array []}
      />
    </ErrorBoundary>
  </Scroll>
</div>
`;
